MarkBind/markbind
# Added List, ListItem, and Media Features - Resolves #906 and #2272

**Overview of changes:**

This PR introduces List, ListItem, and Media features. 
It resolves issue #906 by adding support for custom icons in lists, and also addresses #2272 for additional icon support. 
The new list functionalities enable usage of traditional 'ul' and 'ol' elements, including the incorporation of custom icons from FontAwesome and images.

**Demo 1:**
<img width="329" alt="image" src="https://github.com/MarkBind/markbind/assets/30100720/470e30f8-7ff8-434e-9a65-524e43167be9">
*Syntax*:
```
<List>
  {fas fa-check text-danger} first item
  second item
    {fas fa-check text-success} third item
      {/images/deer.jpg} fourth item
  fifth item
</List>
<List bulletIcon="fas fa-check">
  <ListItem>Custom Icon 1</ListItem>
  <ListItem icon="fas fa-times">Individual Icon 2</ListItem>
  <ListItem>Custom Icon 3</ListItem>
</List>
<List>
  <ListItem icon="fas fa-check text-danger">Individual Icon 1</ListItem>
  <ListItem icon="fas fa-times text-warning">Individual Icon 2</ListItem>
  <ListItem icon="fas fa-exclamation text-success">Individual Icon 3</ListItem>
</List>
```

Demo 2:
<img width="377" alt="image" src="https://github.com/MarkBind/markbind/assets/30100720/76e0f982-0379-4ade-9940-19cec8c520b0">
```
<List bulletStyle="none">
  <ListItem>
    <media 
      src="/images/deer.jpg"
      alt="Example Image 1"
      eager
      imageWidth="100"
      imageHeight="100"
    >
      ## Title 1
      This is some content for the first item.
    </media>
  </ListItem>
  <ListItem>
    <media
      src="/images/deer.jpg"
      alt="Example Image 2"
      eager
      imageWidth="100"
      imageHeight="100"
    >
      ## Title 2
      This is some content for the second item.
    </media>
  </ListItem>
  <ListItem>
    <media
      src="/images/deer.jpg"
      alt="Example Image 3"
      eager
      imageWidth="100"
      imageHeight="100"
    >
      <h2> Title 3 </h2>
      <p> This is some content for the third item. </p>
    </media>
  </ListItem>
</List>
```

Demo 3:
<img width="202" alt="image" src="https://github.com/MarkBind/markbind/assets/30100720/1c866414-85c9-46bd-8d58-44d39c085f18">
```
<List>
  Mark list 1
    Mark list 2
      Mark list 3
  Mark list 4
</List>
<List bulletStyle="disc">
  <ListItem>Disc 1</ListItem>
  <ListItem>Disc 2</ListItem>
  <ListItem>
    Disc 3
    <List bulletStyle="circle">
      <ListItem>Circle 1</ListItem>
      <ListItem>Circle 2</ListItem>
      <ListItem>Circle 3</ListItem>
    </List>
  </ListItem>
</List>
<List bulletStyle="square">
  <ListItem>Square 1</ListItem>
  <ListItem>Square 2</ListItem>
  <ListItem>Square 3</ListItem>
</List>
<List bulletStyle="none">
  <ListItem>None 1</ListItem>
  <ListItem>None 2</ListItem>
  <ListItem>None 3</ListItem>
</List>
```

**Testing instructions:**

1. Ensure that you have the correct branch of this pull request checked out locally.

2. Open a terminal/command prompt and navigate to the **doc** directory of the checked out project.

3. Run the command `markbind serve -d` to start the local server.

4. In the documentation, visit the "Presentation" page to see the newly added sections for "List" and "Media".

Please refer to these sections to see examples and usage of the new features.

Added List, ListItem, and Media features

- [ ] Add support for markdown like syntax
- [ ] Add support for nested list
